src/Microsoft.PowerShell.IoT/Microsoft.PowerShell.IoT.cs

```
@@ -216,20 +246,30 @@ public class SetGpioPin : Cmdlet
 
         protected override void ProcessRecord()
         {
-            if (this.Id != null)
+            try
             {
-                foreach (int pinId in this.Id)
+                if (this.Id != null)
                 {
-                    var pin = Unosquare.RaspberryIO.Pi.Gpio[pinId];
-                    pin.PinMode = Unosquare.RaspberryIO.Gpio.GpioPinDriveMode.Output;
-                    pin.Write((Unosquare.RaspberryIO.Gpio.GpioPinValue)this.Value);
-                    if (this.PassThru)
+                    foreach (int pinId in this.Id)
                     {
-                        GpioPinData pinData = new GpioPinData(pinId, this.Value, pin);
-                        WriteObject(pinData);
+                        var pin = Unosquare.RaspberryIO.Pi.Gpio[pinId];
+                        pin.PinMode = Unosquare.RaspberryIO.Gpio.GpioPinDriveMode.Output;
+                        pin.Write((Unosquare.RaspberryIO.Gpio.GpioPinValue)this.Value);
+                        if (this.PassThru)
+                        {
+                            GpioPinData pinData = new GpioPinData(pinId, this.Value, pin);
+                            WriteObject(pinData);
+                        }
                     }
                 }
             }
+            catch (System.TypeInitializationException e) // Unosquare.RaspberryIO.Gpio.GpioController.Initialize throws this TypeInitializationException
+            {
+                if (!Unosquare.RaspberryIO.Computer.SystemInfo.Instance.IsRunningAsRoot)
+                {
+                    throw new PlatformNotSupportedException(Resources.ErrNeedRootPrivileges, e);
+                }
+            }
```

If they are running as root we should probably `throw` the error again so we don't accidentally swallow it?